monoio/src/utils/thread_id.rs

```diff
@@ -3,7 +3,13 @@ use std::sync::{
     LazyLock,
 };
 
-static ID_GEN: LazyLock<AtomicUsize> = LazyLock::new(|| AtomicUsize::new(1));
+// thread id begins from 16.
+// 0 is always reserved
+// 1 is blocking thread
+// 2-15 are unused
+static ID_GEN: LazyLock<AtomicUsize> = LazyLock::new(|| AtomicUsize::new(16));
+
+pub(crate) const BLOCKING_THREAD_ID: usize = 1;
 
 /// Used to generate thread id.
 pub(crate) fn gen_id() -> usize {
```

Comment:
Use a relax ordering for incrementing-id?

Reply:
Here we need to get old value, Relaxed may cause we get the same thread id?

Reply:
The relax ordering only affects the happen-before behaver between read and write with different threads.
Here the read after write with syncing in same thread.
I think just use relaxed ordering is OK

Reply:
More: https://en.cppreference.com/w/cpp/atomic/memory_order#Relaxed_ordering

Reply:
Reddit: https://www.reddit.com/r/rust/comments/aeii1i/is_memory_orderingrelaxed_sufficient_to_generate/